src/types.rs

```diff
@@ -148,6 +148,32 @@ impl serde::Deserialize for Sectors {
     }
 }
 
+// A type for Data Blocks as used by the thin pool.
+custom_derive! {
+    #[derive(NewtypeFrom, NewtypeAdd, NewtypeSub, NewtypeDeref,
```

Comment:
Most of these ```Newtype*``` macro calls should be deleted. It is probably possible to use the same subset as are used for Bytes and Sectors. It would probably be fine to use the ```unsigned_mul``` macros, although I can't see anywhere that DataBlocks are multiplied by any unsigned numbers.

AFAICT, the size of a DataBlock in Sectors is not a constant? In that case, you're probably better off parameterizing the DataBlock on its scaling factor in Sectors.